src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -411,4 +405,67 @@ private void deleteFileAndInvokeCallback(
         mTotalSize = DiskBasedCacheUtility.removeEntry(key, mTotalSize, mEntries);
         callback.onGetComplete(null);
     }
+
+    /**
+     * Builder is used to build an instance of {@link DiskBasedAsyncCache} from values configured by
+     * the setters.
+     */
+    public static class Builder {
+        @Nullable private FileSupplier rootDirectorySupplier;
+        @Nullable private File rootDirectory;
+        private int maxCacheSizeInBytes;
+
+        public Builder() {
+            rootDirectorySupplier = null;
+            rootDirectory = null;
+            maxCacheSizeInBytes = -1;
```

Comment:
nit: what do you think of initializing those values at the declaration? This would be more concise, as we could remove the constructor

Reply:
I like that